frontend/src/components/readsets/ReadsetsListContent.tsx

```diff
@@ -1,105 +1,279 @@
 import { FILTER_TYPE } from "../../constants";
 import { setColumnWidths } from "../pagedItemsTable/tableColumnUtilities";
-import { setReleaseStatusAll } from "../../modules/datasets/actions";
-import { setReleaseStatus } from "../../modules/readsets/actions";
+import { ReleaseStatus } from "../../models/fms_api_models";
+import { ExpandableConfig } from "antd/lib/table/interface";
+import api from "../../utils/api";
+import produce from "immer";
+import { useCurrentUser } from "../../hooks/useCurrentUser";
+
+const RELEASE_STATUS_STRING = {
+    [ReleaseStatus.RELEASED]: 'Released',
+    [ReleaseStatus.BLOCKED]: 'Blocked',
+    [ReleaseStatus.AVAILABLE]: 'Available',
+} as const
+const OPPOSITE_STATUS = {
+    [ReleaseStatus.AVAILABLE]: ReleaseStatus.RELEASED,
+    [ReleaseStatus.RELEASED]: ReleaseStatus.BLOCKED,
+    [ReleaseStatus.BLOCKED]: ReleaseStatus.RELEASED,
+} as const
 
 interface ReadsetsListContentProps {
-    dataset?: Dataset
-    laneValidationStatus?: ValidationStatus
+    dataset: Dataset
+    laneValidationStatus: ValidationStatus
+    refreshDataset: () => Promise<void>
 }
 
-const wrapReadset = (readset: Readset): ObjectWithReadset => {
-    return { readset }
-}
-const ReadsetsListContent = ({ dataset, laneValidationStatus }: ReadsetsListContentProps) => {
-    const readsetsByID = useAppSelector(selectReadsetsByID)
+const ReadsetsListContent = ({ dataset, laneValidationStatus, refreshDataset }: ReadsetsListContentProps) => {
+    const user = useCurrentUser()
+    const isAdmin = user ? user.is_staff : false
+
     const readsetTableState = useAppSelector(selectReadsetsTable)
     const readsetTableCallbacks = usePagedItemsActionsCallbacks(ReadsetTableActions)
     const dispatch = useAppDispatch()
-    const { filters, totalCount } = readsetTableState
-
-    const allFilesReleased = dataset?.released_status_count === totalCount
-    const allFilesBlocked = dataset?.blocked_status_count === totalCount
+    const { filters } = readsetTableState
 
     // For this table, there is a single PagedItems redux state. We need to reset the paged items
     // state whenever a new dataset is selected by the user. This component keeps track of the
     // last dataset that was loaded so that it can detect when it has received a new dataset
     // and has to flush the state.
-    const currentDatasetID = dataset ? +dataset.id : undefined
     useEffect(() => {
-        if (currentDatasetID) {
-            dispatch(ReadsetTableActions.resetPagedItems())
-            dispatch(ReadsetTableActions.setFixedFilter(createFixedFilter(FILTER_TYPE.INPUT_OBJECT_ID, 'dataset__id', String(currentDatasetID))))
-            dispatch(ReadsetTableActions.listPage(1))
+        dispatch(ReadsetTableActions.resetPagedItems())
+        dispatch(ReadsetTableActions.setFixedFilter(createFixedFilter(FILTER_TYPE.INPUT_OBJECT_ID, 'dataset__id', dataset.id.toString())))
+        dispatch(ReadsetTableActions.listPage(1))
```

Comment:
careful with this piece of code, i believe this function launches a second api call to fetch the current page list